gui.py

```diff
@@ -75,19 +76,45 @@ def auth(self, EMAIL_ACCOUNT):
 			print "Mailboxes located."
 
 		rv, data = M.select(EMAIL_FOLDER)
-
 		if rv == 'OK':
 			print "Now processing mailbox ...\n"
+			print 'search mail : ', search_email, 'Current dir : ',curr_dir
 			self.process_mailbox(M, search_email, curr_dir)
 			M.close()
 		else:
 			print "ERROR: Unable to open mailbox ", rv
 
 		M.logout()
 
+	def validate(self, date_text):
+		day,month,year = date_text.split('/')
+		isValidDate = True	
+		try :
+			datetime(int(year),int(month),int(day))
+		except ValueError :
+			isValidDate = False
+		if(isValidDate) :
+			return True
+		else :
+			return False
+
 	def process_mailbox(self, M, search_email, curr_dir):
 
-		rv, data = M.search(None, '(OR (TO %s) (FROM %s))' % (search_email, search_email))
+		fromdate = self.ui.lineEdit_4.text()
+		todate = self.ui.lineEdit_5.text()
+		#print 'date from and to ',fromdate, todate
+		
+		if self.validate(fromdate) and self.validate(todate):
+			fromdate = datetime.strptime(str(fromdate), '%d/%m/%Y')
+			fromdate = fromdate.strftime('%d-%b-%Y')
+			todate = datetime.strptime(str(todate), '%d/%m/%Y')
+			todate = todate.strftime('%d-%b-%Y')
+			
+			rv, data = M.search(None, '(OR (TO %s) (FROM %s) (SINCE %s BEFORE %s))' % (search_email, search_email, fromdate, todate))
+			#rv, data = M.search(None, '(OR (TO %s) (FROM %s) (SINCE "12-JAN-2017" BEFORE "12-DEC-2017"))' % (search_email, search_email))
```

Comment:
remove the comments

Reply:
Sure! and should I then send a PR again ?